woocommerce/woocommerce-ios
# [REST API] Introduce A/B test and remove feature flag

Part of: #8679


### Description
This PR adds a new A/B test experiment for the REST API project to enable application password authentication. Experiment details: pbxNRc-2i4-p2

- Added new A/B test experiment.
- Replaced the temporary feature flag with A/B test.

### Testing instructions

#### Testing the `control` variation

- Log out of the app or skip login onboarding if needed.
- On the prologue screen, observe that you see the login screen as before.
- Login flow should work as before without any changes.

#### Testing the `treatment` variation

We will have to sandbox and assign a particular variation to the device to test the variations.

Prerequisites:
- Please follow the setup in PCYsg-Fq7-p2#assignments-api for proxied sandboxed API requests: connect to sandbox via ssh, and point WP.com API to the sandbox IP address (`ifconfig`) in the local matchine's `/etc/hosts`
- In Xcode, set a breakpoint at `ExPlatService` L47 where `anon_id` is set in the URL request
- Make an API request `PATCH /wpcom/v2/experiments/0.1.0/assignments` with the following `body` and make sure it's successful:
```
{
    "variations": {
        "woocommerceios_login_rest_api_project_202301_v2": "treatment"
    },
    "anon_id": "anon_id_from_the_previous_step",
    "username_override": ""
}
```
- Launch the app, and let it sit for a bit
- Close and relaunch the app 

#### 1. Prologue screen and site address login flow.
- Log out of the app or skip onboarding if needed.
- Notice that the prologue screen has only CTAs to log in with the site address and create a new store.
- Select Enter your site address and proceed with the address of a self-hosted site. Notice that the app will then navigate to the site credential login screen immediately.
- Try again with the address of a WPCom site. Tapping Continue should navigate to WPCom login flow.

#### 2. Application password disabled screen
Pre-requisite: Disable application password on your self-hosted site (e.g. by installing [WordFence plugin](https://www.wordfence.com/) Free version is enough for testing).
- Log out of the app or skip onboarding if needed.
- Notice that the prologue screen has only CTAs to log in with site address and create a new store.
- Select Enter your site address and proceed with the address of a self-hosted site.
- Enter your site credentials. You should then be navigated to the application password disabled screen.
- Notice that there's a primary button on the screen to log in with WPCom. Tap that button.
- Continue with the WPCom login flow. After the login succeeds, you should be able to navigate to the My Store screen.

### Screenshots

| | Control | Treatment |
| -----| ----- | ----- |
| Prologue screen | <image src="https://user-images.githubusercontent.com/5533851/213160111-3f6c0fec-2dc8-402e-8b8a-5dd5bda07ebc.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/213160207-d5fa8774-7407-4e36-a9dd-30438c9e1d59.png" width=320 /> |
| After enter site address | <img src="https://user-images.githubusercontent.com/5533851/213160340-357743a9-48a0-42b5-b368-9ec7f10cdc85.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/213160419-0759768d-f09c-441d-ada5-519fb789ed61.png" width=320 /> |

Application password disabled:
<img src="https://user-images.githubusercontent.com/5533851/213163507-196a86df-5985-4a5b-8d4e-2d41a27d1266.png" width=320 />

---
